### PR TITLE
added grpc services and tests for them

### PR DESCRIPTION
**Pull request by OlehSarapuk**

### CODE REVIEW

File: src/test/java/com/example/ajaxonboardingproject/NatsMovieSessionControllerTests.kt [hunk 1]
```diff
@@ -32,19 +32,18 @@ class NatsMovieSessionControllerTests {
...
             NatsSubject.ADD_NEW_MOVIE_SESSION_SUBJECT,
-            request.toByteArray(),
+            expected.toByteArray(),
             Duration.ofMillis(100000)
         )
         //Then
-        val reply = MovieSessionResponse.parseFrom(future.get().data)
-        assertThat(request.movieSession).isEqualTo(reply.movieSession)
+        val actual = MovieSessionResponse.parseFrom(future.get().data)
+        assertThat(expected.movieSession).isEqualTo(actual.movieSession)
```
Comment (ajax-gnatiuk-s) [inline src/test/java/com/example/ajaxonboardingproject/NatsMovieSessionControllerTests.kt]:
```
assertThat(<ACTUAL>).isEqualTo(<EXPECTED>)
```

File: src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    override fun getAllMovies(
+        request: MovieRequest,
+        responseObserver: StreamObserver<MovieResponse>
+    ) {
+        movieService.getAll()
+            .map { movieConverter.movieToProtoResponse(it) }
+            .collectList()
+            .block()!!
+            .forEach { responseObserver.onNext(it) }
```
Comment (ajax-gnatiuk-s) [inline src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt]:
```suggestion
        movieService.getAll()
            .map { movieConverter.movieToProtoResponse(it) }
            .doOnNext { responseObserver.onNext(it) }
            .blockLast()
```

?

File: src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private val movieConverter: MovieConverter
+): MovieServiceGrpc.MovieServiceImplBase() {
+
+    override fun addMovie(
+        request: MovieRequest,
+        responseObserver: StreamObserver<MovieResponse>
+    ) {
+        val movie = movieService.add(movieConverter.protoRequestToMovie(request)).block()!!
+        responseObserver.onNext(movieConverter.movieToProtoResponse(movie))
+        responseObserver.onCompleted()
```
Comment (ajax-gnatiuk-s) [inline src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt]:
```suggestion
        movieService.add(movieConverter.protoRequestToMovie(request))
            .doOnNext { responseObserver.onNext(movieConverter.movieToProtoResponse(it)) }
            .block()
        responseObserver.onCompleted()
```

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

**Review by ajax-gnatiuk-s — APPROVED** [review]

File: src/test/java/com/example/ajaxonboardingproject/NatsMovieSessionControllerTests.kt (hunk 1 above)
Comment (OlehSarapuk) [reply src/test/java/com/example/ajaxonboardingproject/NatsMovieSessionControllerTests.kt]: Done! ✅

File: src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt (hunk 2 above)
Comment (OlehSarapuk) [reply src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt]: Done! ✅

File: src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt (hunk 3 above)
Comment (OlehSarapuk) [reply src/main/java/com/example/ajaxonboardingproject/nats/grpc/MovieGrpcService.kt]: Done! ✅

**Review by Wectro20 — APPROVED** [review]

**Pull request merged**